src/Samhammer.AzureBlobStorage/Services/StreamManagerService.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.IO;
+using Microsoft.Extensions.Options;
+using Microsoft.IO;
+using Samhammer.AzureBlobStorage.Options;
+
+namespace Samhammer.AzureBlobStorage.Services
+{
+    public class StreamManagerService : IStreamManagerService
+    {
+        private const long MaxSmallPoolFreeBytes = 1000000;
+
+        private const long MaxLargePoolFreeBytes = 10000000;
+
+        private readonly RecyclableMemoryStreamManager _streamManager;
+
+        public StreamManagerService(IOptions<StreamManagerOptions> streamManagerOptions)
+        {
+            StreamManagerOptions options = streamManagerOptions.Value;
+            _streamManager = new RecyclableMemoryStreamManager(
+                options.MaxSmallPoolFreeBytes.HasValue ? options.MaxSmallPoolFreeBytes.Value : MaxSmallPoolFreeBytes,
+                options.MaxLargePoolFreeBytes.HasValue ? options.MaxLargePoolFreeBytes.Value : MaxLargePoolFreeBytes);
```

Comment:
@khoanguyen102 you could simplify the fallback code here :) resharper suggests this

```
            _streamManager = new RecyclableMemoryStreamManager(
                options.MaxSmallPoolFreeBytes ?? MaxSmallPoolFreeBytes,
                options.MaxLargePoolFreeBytes ?? MaxLargePoolFreeBytes);
```

Reply:
Thanks @agentschmitt, I have updated it my Resharper license was expired :(. Let me find another one